hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -87,21 +88,20 @@ public static void cascade(
 			final boolean hasUninitializedLazyProperties = persister.hasUninitializedLazyProperties( parent );
 			final int componentPathStackDepth = 0;
 			for ( int i = 0; i < types.length; i++) {
-				final CascadeStyle style = cascadeStyles[i];
-				final String propertyName = propertyNames[i];
-
-				if ( style.doCascade( action ) ) {
-					Object child;
+				final CascadeStyle style = cascadeStyles[ i ];
+				final String propertyName = propertyNames[ i ];
 
+				Object child = null;
+				if ( style.doCascade( action ) || action.deleteOrphans() ) {
 					if ( hasUninitializedLazyProperties &&
 							!persister.getInstrumentationMetadata().isAttributeLoaded( parent, propertyName ) ) {
 						// parent is a bytecode enhanced entity.
 						// cascading to an uninitialized, lazy value.
-						if ( types[i].isCollectionType() ) {
+						if ( types[ i ].isCollectionType() ) {
 							// The collection does not need to be loaded from the DB.
 							// CollectionType#resolve will return an uninitialized PersistentCollection.
 							// The action will initialize the collection later, if necessary.
-							child = types[i].resolve( LazyPropertyInitializer.UNFETCHED_PROPERTY, eventSource, parent );
+							child = types[ i ].resolve( LazyPropertyInitializer.UNFETCHED_PROPERTY, eventSource, parent );
 							// TODO: it would be nice to be able to set the attribute in parent using
 							// persister.setPropertyValue( parent, i, child ).
 							// Unfortunately, that would cause the uninitialized collection to be
```

Comment:
On line 110 is:

`else if ( action.performOnLazyProperty() ) {
 	// The (non-collection) attribute needs to be initialized so that
 	// the action can be performed on the initialized attribute.
 	LazyAttributeLoadingInterceptor interceptor = persister.getInstrumentationMetadata().extractInterceptor( parent );
 	child = interceptor.fetchAttribute( parent, propertyName );
}`

If the attribute is uninitialized and the action does not cascade, then there should be no reason to initialize the attribute. Since the attribute is uninitialized, then there is no need to delete orphans.

The if statement should probably be:

`else if ( style.doCascade() && action.performOnLazyProperty() ) {`
 	...
}`
The code